lib/Storage/Google.php

```diff
@@ -74,6 +75,10 @@ public function __construct($params) {
 			$this->service = new \Google_Service_Drive($this->client);
 			$token = json_decode($params['token'], true);
 			$this->id = 'google::'.substr($params['client_id'], 0, 30).$token['created'];
+			$this->root = isset($params['root']) ? $params['root'] : '';
+			if(!$this->is_dir('')) {
```

Comment:
Please don't access the storage inside the constructor as it would slow down other bits of ownCloud. Storages are currently instantiated even when not used.

I suggest removing this check completely. I expect that if the root is wrong it will already throw an exception during verification when the function `test()` of the storage is called.